packages/data/addon/services/keg.js

```diff
@@ -80,30 +74,30 @@ export default Service.extend({
    * @returns {Array} records that were pushed to the keg
    */
   pushMany(type, rawRecords, options = {}) {
-    let factory = this._getFactoryForType(options.modelFactory || type),
-      recordKeg = this._getRecordKegForType(type),
-      idIndex = this._getIdIndexForType(type),
-      identifierField = get(factory, 'identifierField') || get(this, 'identifierField');
-
-    let returnedRecords = A();
+    const factory = this._getFactoryForType(options.modelFactory || type);
+    const recordKeg = this._getRecordKegForType(type);
+    const idIndex = this._getIdIndexForType(type);
+    const identifierField = factory.identifierField || KegService.identifierField;
+    const owner = getOwner(this);
 
+    const returnedRecords = A();
     for (let i = 0; i < rawRecords.length; i++) {
```

Comment:
Could we turn this into a map function?

Reply:
Since the record length can be large, `for` loops are faster: https://jsperf.com/for-loop-vs-map-vs-foreach